synthtool/gcp/templates/java_library/CONTRIBUTING.md

```diff
@@ -25,4 +25,109 @@ information on using pull requests.
 ## Community Guidelines
 
 This project follows
-[Google's Open Source Community Guidelines](https://opensource.google.com/conduct/).
\ No newline at end of file
+[Google's Open Source Community Guidelines](https://opensource.google.com/conduct/).
+
+## Building the project
+
+To build, package and run all unit tests simple run the command
+
+```
+mvn clean verify
+```
+
+### Running Integration tests
+
+To include integration tests when building the project you will need access to
+a GCP Project with a valid service account. 
+
+For instructions on how to generate a service account and corresponding
+credentials JSON see: [Creating a Service Account][1].
+
+Then run the following to build, package, run all unit tests and run all
+integration tests.
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service/account.json
+mvn -Penable-integration-tests clean verify
+```
+
+## Code Samples
+
+Code Samples must be authored in their own Maven modules, and guarded by a
+Maven profile by the name `enable-samples`.
+
+The samples must be separate from the primary project for a few reasons:
+1. Primary projects have a minimum Java version of Java 7 whereas samples have
+   a minimum Java version of Java 8. Due to this we need the ability to
+   selectively exclude samples from a build run.
+2. Many code samples are often dependent on external GCP services and need
+   credentials to access the service.
+3. At this time, code samples are not released as Maven artifacts and must be
+   able to be excluded from any release builds we run.
+   
+### Building
+
+To build and run tests for samples [activate](#profile-activation) the profile
+and run
+
+```bash
+mvn clean verify
+```
+
+Some samples require access to GCP services and require a service account
+
+```bash
+export GOOGLE_APPLICATION_CREDENTIALS=/path/to/service/account.json
+mvn clean verify
+```
+
+### Profile Config
+
+1. To add samples in a profile to your Maven project add the following to your
+`pom.xml`
+
+    ```xml
+    <project>
+      [...]
+      <profiles>
+        <profile>
+          <id>enable-samples</id>
+          <modules>
+            <module>sample</module>
+          </modules>
+        </profile>
+      </profiles>
+      [...]
+    </project>
+    ```
+
+2. [Activate](#profile-activation) the profile.
```

Comment:
you don't need to activate now. That's done at runtime. This step can be deleted.

Reply:
Activation can be done via `settings.xml` so that it's not needed on the command line, this leaves the option for folks to pick which activation they want to use.